src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/ReactiveCourseRepositoryIT.java

```diff
@@ -0,0 +1,197 @@
+    private static final Course COURSE_5 = new Course(COURSE_ID_5, COURSE_NAME_5, DEPARTMENT_NAME_1);
+
+    private final CosmosEntityInformation<Course, String> entityInformation =
+        new CosmosEntityInformation<>(Course.class);
+
+    @Autowired
+    private ReactiveCosmosTemplate template;
+
+    @Autowired
+    private ReactiveCourseRepository repository;
+
+    @PreDestroy
+    public void cleanUpCollection() {
+        template.deleteContainer(entityInformation.getCollectionName());
+    }
+
+    @Before
+    public void setup() {
+        final Flux<Course> savedFlux = repository.saveAll(Arrays.asList(COURSE_1, COURSE_2,
+            COURSE_3, COURSE_4));
+        StepVerifier.create(savedFlux).thenConsumeWhile(course -> true).expectComplete().verify();
+    }
+
+    @After
+    public void cleanup() {
+        final Mono<Void> deletedMono = repository.deleteAll();
+        StepVerifier.create(deletedMono).thenAwait().verifyComplete();
+    }
+
+    @Test
+    public void testFindById() {
+        final Mono<Course> idMono = repository.findById(COURSE_ID_4);
+        StepVerifier.create(idMono).expectNext(COURSE_4).expectComplete().verify();
+    }
+
+    @Test
+    public void testFindByIdAndPartitionKey() {
+        final Mono<Course> idMono = repository.findById(COURSE_ID_4,
+            new PartitionKey(entityInformation.getPartitionKeyFieldValue(COURSE_4)));
+        StepVerifier.create(idMono).expectNext(COURSE_4).expectComplete().verify();
+    }
+
+    @Test
+    public void testFindByIdAsPublisher() {
+        final Mono<Course> byId = repository.findById(Mono.just(COURSE_ID_1));
+        StepVerifier.create(byId).expectNext(COURSE_1).verifyComplete();
+    }
+
+    @Test
+    public void testFindAllWithSort() {
+        final Flux<Course> sortAll = repository.findAll(Sort.by(Sort.Order.desc("name")));
+        StepVerifier.create(sortAll).expectNext(COURSE_4, COURSE_3, COURSE_2, COURSE_1).verifyComplete();
+    }
+
+    @Test
+    public void testFindByIdNotFound() {
+        final Mono<Course> idMono = repository.findById("10");
+        //  Expect an empty mono as return value
+        StepVerifier.create(idMono).expectComplete().verify();
+    }
+
+    @Test
+    public void testFindByIdAndPartitionKeyNotFound() {
+        final Mono<Course> idMono = repository.findById("10",
+            new PartitionKey(entityInformation.getPartitionKeyFieldValue(COURSE_1)));
+        //  Expect an empty mono as return value
+        StepVerifier.create(idMono).expectComplete().verify();
+    }
+
+    @Test
+    public void testFindAll() {
+        final Flux<Course> allFlux = repository.findAll();
+        StepVerifier.create(allFlux).expectNextCount(4).verifyComplete();
+    }
+
+    @Test
+    public void testInsert() {
+        final Mono<Course> save = repository.save(COURSE_5);
+        StepVerifier.create(save).expectNext(COURSE_5).verifyComplete();
+    }
+
+    @Test
+    public void testUpsert() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/Microsoft/spring-data-cosmosdb/pullRequest?prid=4741022)